WolvenKit.App/Helpers/ProjectResourceTools.cs

```diff
@@ -358,141 +361,208 @@ public async Task MoveAndRefactor(Cp77Project activeProject, string sourcePath,
-                        "Directory already exists!",
-                        WMessageBoxButtons.YesNoCancel);
-                    overwriteFiles = response is WMessageBoxResult.Yes;
-                    if (response is WMessageBoxResult.Cancel)
-                    {
-                        return;
-                    }
-                }
+                continue;
+            }
 
-                FileHelper.MoveRecursively(sourceAbsPath, destAbsPath, overwriteFiles, projectRootPath, _loggerService);
+            await MoveFileAsync(sourceAbsPath, targetAbsPath, activeProject, sourceIsDirectory);                
+        }
 
-                var sourceInRaw = sourceAbsPath.Replace("archive", "raw");
-                if (sourceInRaw != sourceAbsPath && Directory.Exists(sourceInRaw))
-                {
-                    FileHelper.MoveRecursively(sourceInRaw, destAbsPath.Replace("archive", "raw"), overwriteFiles, projectRootPath,
-                        _loggerService);
-                }
+        // Delete only files that were successfully replaced
+        var successfulReplacements = fileReplacements.Where((kvp) => File.Exists(kvp.Value)).ToDictionary();
 
-                _loggerService.Info($"Moved \"{sourceRelPath}{Path.DirectorySeparatorChar}*\" to \"{destRelPath}\"");
-            }
+        foreach (var originalFile in successfulReplacements.Keys.Where(File.Exists))
+        {
+            File.Delete(originalFile);
+        }
 
-            catch (Exception e)
-            {
-                _loggerService.Error(
-                    $"Error when trying to move \"{sourceRelPath}{Path.DirectorySeparatorChar}*\": {e.Message}");
-            }
+        // Delete empty directories
+        if (successfulReplacements.Count > 0)
+        {
+            DeleteEmptyDirectoriesRecursive(sourceFileOrDirAbsPath);
         }
 
         if (!refactor)
         {
             return;
         }
 
-     
-        var newRelPath = activeProject.GetResourcePathFromRoot(destAbsPath);
+        ReplacePathInProject(activeProject, successfulReplacements);
+    }
+
+    private static void MoveDirectoryAndChildren(string sourceAbsPath, string destAbsPath)
+    {
+        if (!Directory.Exists(sourceAbsPath))
+        {
+            return;
+        }
 
-        ReplacePathInProject(activeProject, sourceAbsPath, newRelPath);
+        if (!Directory.Exists(destAbsPath))
+        {
+            Directory.CreateDirectory(destAbsPath);
+        }
+
+        foreach (var directory in Directory.GetDirectories(sourceAbsPath, "*", SearchOption.AllDirectories))
+        {
+            var relativePath = Path.GetRelativePath(sourceAbsPath, directory);
+            var targetDirPath = Path.Combine(destAbsPath, relativePath);
+
+            Directory.CreateDirectory(targetDirPath);
+        }
+
+        Directory.Delete(sourceAbsPath, true);
+    }
+
+    private static void DeleteEmptyDirectoriesRecursive(string directoryPath)
```

Comment:
Should this function 
1. delete all directories recursively if the whole tree is only empty directories or
2. delete only the empty directories of a tree that may also include non-empty directories?

Reply:
This should delete everything that is an empty directory, but nothing that isn't